snapcraft/plugins/dotnet.py

```diff
@@ -58,11 +52,28 @@ class DotNetPlugin(snapcraft.BasePlugin):
     def schema(cls):
         schema = super().schema()
 
+        schema['properties']['dotnet-runtime-version'] = {
+            'type': 'string',
+            'default': _RUNTIME_DEFAULT,
+        }
+
         if 'required' in schema:
             del schema['required']
 
         return schema
 
+    @classmethod
+    def get_pull_properties(cls):
+        # Inform Snapcraft of the properties associated with pulling. If these
+        # change in the YAML Snapcraft will consider the build step dirty.
+        return ['dotnet-runtime-version']
+
+    @classmethod
+    def get_build_properties(cls):
+        # Inform Snapcraft of the properties associated with building. If these
+        # change in the YAML Snapcraft will consider the build step dirty.
+        return ['dotnet-runtime-version']
```

Comment:
There's no need to specify both of these-- if they effect both, just put them in `get_pull_properties`. Marking the pull step dirty marks the build step dirty as well.